docs/speech-usage.rst

```diff
@@ -30,11 +31,41 @@ create an instance of :class:`~google.cloud.speech.client.Client`.
      >>> client = speech.Client()
 
 
+Asychronous Recognition
+-----------------------
+
+The :meth:`~google.cloud.speech.Client.async_recognize` sends audio data to the
+Speech API and initiates a Long Running Operation. Using this operation, you
+can periodically poll for recognition results. Use asynchronous requests for
+audio data of any duration up to 80 minutes.
+
+See: `Speech Asynchronous Recognize`_
+
+
+  .. code-block:: python
+
+      >>> import time
+      >>> operation = client.async_recognize(
+      ...     None, 'gs://my-bucket/recording.flac',
+      ...     'FLAC', 16000, max_alternatives=2)
+      >>> retry_count = 100
+      >>> while retry_count > 0 and not operation.complete:
+      ...     retry_count -= 1
+      ...     time.sleep(10)
+      ...     operation.poll()  # API call
+      >>> operation.complete
+      True
+      >>> operation.results[0]['alternatives'][0]['transcript']
```

Comment:
@tseaver, I can do that just like [`sync_recognize`](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/b7f004e00f1ee021b0cb7ac9261fcc6c021dfacb/speech/google/cloud/speech/client.py#L180) but then we'll have to update it later for batch handling.